app/javascript/packs/suspendUserModal.js

```diff
@@ -0,0 +1,103 @@
+import { showWindowModal } from '@utilities/showModal';
+import { request } from '@utilities/http';
+
+const suspendUser = async ({ event, userId, username, suspensionReason }) => {
+  event.preventDefault();
+  closeSuspendUserModal();
+
+  try {
+    const response = await request(
+      `/admin/member_manager/users/${userId}/user_status`,
+      {
+        method: 'PATCH',
+        body: JSON.stringify({
+          id: userId,
+          user: {
+            note_for_current_role: suspensionReason,
+            user_status: 'Suspend',
+          },
+        }),
+        credentials: 'same-origin',
+      },
+    );
+
+    const outcome = await response.json();
+
+    /* eslint-disable no-restricted-globals */
+    if (outcome.success) {
+      top.addSnackbarItem({
+        message: `Success: "${username}" has been suspended.`,
+        addCloseButton: true,
+      });
+    } else {
+      top.addSnackbarItem({
+        message: `Error: something went wrong; ${username} NOT suspended.`,
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+};
+
+function closeSuspendUserModal() {
+  window.parent.document
+    .querySelector('#window-modal .crayons-modal__dismiss')
+    .click();
+}
+
+const modalContents = new Map();
+
+function getModalContents(modalContentSelector) {
+  if (!modalContents.has(modalContentSelector)) {
+    const modalContentElement =
+      window.parent.document.querySelector(modalContentSelector);
+    const modalContent = modalContentElement.innerHTML;
+
+    modalContentElement.remove();
+    modalContents.set(modalContentSelector, modalContent);
+  }
+
+  return modalContents.get(modalContentSelector);
+}
+
+function checkSuspensionReason(event) {
+  const { userId, username, suspensionReasonSelector } = event.target.dataset;
+  const suspendUserModal =
+    window.parent.document.getElementById('window-modal');
+  const suspensionReason = suspendUserModal.querySelector(
+    suspensionReasonSelector,
+  ).value;
+
+  if (!suspensionReason) {
+    suspendUserModal
+      .querySelector('#suspension-reason-error')
+      .classList.remove('hidden');
```

Comment:
At the moment, screen reader users wouldn't be aware of this message appearing (and therefore might be confused about why their form isn't submitting).

Could we set the text here instead of toggling the `hidden` class? This would let us make the message accessible via an `aria-live` region
```suggestion
    suspendUserModal
      .querySelector('#suspension-reason-error')
      .innerText = "You must give a suspension reason"
```